blocks/fast_reader_test.go

```diff
@@ -36,33 +47,57 @@ func TestFastRead(t *testing.T) {
 	}
 
 	for i, test := range tests {
```

Comment:
For now, I've just updated the original test. If we want to use this PR, I suggest:

 * adding the race test as a new `TestFastReadPool` test case;
 * adding a mise "test" task that runs all the tests with race detection on.